### PR TITLE
Watching imported files

### PR DESCRIPTION
**Pull request by krasimir**
## Why

This PR is addressing https://github.com/zeit/now-cli/issues/2370 where `now dev` is not watching for changes in the files which are imported. For example:

```js
// now.json
{
  "version": 2,
  "builds": [
    { "src": "api/ping.js", "use": "@now/node" }
  ],
  "routes": [
    { "src": "/api/ping", "dest": "/api/ping.js" }
  ]
}

// api/ping.js
import { status } from './data';

const JSONResponse = res => (data, status = 200) => {
...
};

export default (req, res) => {
  return JSONResponse(res)({ status: status() });
};

// api/data.js
export function status() {
  return 'foobar';
}
```

If I make a change in `api/ping.js` there is a rebuild happening. While if I change `api/data.js` nothing happens. This is an issue because we have to kill `now dev` process and start it again in order to get a rebuild.

## What

This PR extends `BuildMatch` type with a `dependencyTree` field which contains all the dependencies of the build entry point. The problem of `dev-server` is not that it can't recognize the change. `chokidar` does that successfully. It is the fact that we are matching the change only agains the build's entry point. The `dependency-tree` module tracks successfully the dependencies and generates an array of strings which we can match against.
The dependency tree is updated at the same time as the matched builders. I tested it for a couple of hours and it seems stable.

### CODE REVIEW

**Comment by TooTallNate** [comment]
Hey @krasimir, thanks for the PR. However there's a fundamental issue with the implementation in that it would only work with the `@now/node` builder.

In fact, we already have a more generic mechanism to enable watching of imported files as part of the Builders API, but we have not yet implemented it in `@now/node` (there is an [in-progress PR](https://github.com/zeit/now-builders/pull/400)).

In short, the `build()` function in the builder needs to return an array of files to watch in the `watch` array. See the PR for the `@now/go` builder for an example: https://github.com/zeit/now-builders/pull/412

**Comment by krasimir** [comment]
Ah I see. Fair enough, closing this one ;)
